[WIP] Fix cudnn channels_last descriptio problem

VitalyFedyunin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #31633 [WIP Patched] Cudnn grouped convolution nhwc patch
* #31428 WIP
* **#31139 [WIP] Fix cudnn channels_last descriptio problem**
* #31131 Add memory_format support to `zeros`, `ones`, `full`

Differential Revision: [D18959229](https://our.internmc.facebook.com/intern/diff/D18959229)

VitalyFedyunin committed:
[WIP] Fix cudnn channels_last descriptio problem

[ghstack-poisoned]

kostmo commented:
## CircleCI build failures summary
As of commit 4f75285:
* **1/8** broken upstream at merge base f73cd28 (see [grid view](https://dr.pytorch.org/master-timeline.html?count=1&sha1=f73cd280827b78991fe95366e823cfadb3f00ebc&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
    * You may want to rebase on the `viable/strict` branch (see its [recency history](https://dr.pytorch.org/master-viable-commits.html)):
        * Since your merge base is older than `viable/strict`, run these commands:
          ```
          git fetch viable/strict
          git rebase viable/strict
          ```
* **7/8** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=4f7528546bfd43b736b36311284ace9d29cbb061).

### 7 new failures recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3908023) pytorch_linux_xenial_py3_6_gcc5_4_build (1/7)
**Step:** "Build" ([details](https://dr.pytorch.org/build-details.html?build_id=72482302))
```
CONFLICT (add/add): Merge conflict in .circleci/config.yml
Auto-merging .circleci/config.yml
CONFLICT (add/add): Merge conflict in .circleci/cimodel/data/pytorch_build_data.py
Auto-merging .circleci/cimodel/data/pytorch_build_data.py
CONFLICT (add/add): Merge conflict in .circleci/cimodel/data/dimensions.py
Auto-merging .circleci/cimodel/data/dimensions.py
CONFLICT (add/add): Merge conflict in .circleci/cimodel/data/caffe2_build_data.py
Auto-merging .circleci/cimodel/data/caffe2_build_data.py
CONFLICT (add/add): Merge conflict in .circleci/cimodel/data/binary_build_data.py
Auto-merging .circleci/cimodel/data/binary_build_data.py
Automatic merge failed; fix conflicts and then commit the result.
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3910525) caffe2_onnx_py2_gcc5_ubuntu16_04_test (2/7)
**Step:** "Test" ([details](https://dr.pytorch.org/build-details.html?build_id=72615438))
```
Dec 11 23:29:33 test/onnx/test_operators.py::TestOperators::test_empty_like PASSED       [  2%]
Dec 11 23:29:33 test/onnx/test_operators.py::TestOperators::test_empty_like_opset7 PASSED [  2%]
Dec 11 23:29:33 test/onnx/test_operators.py::TestOperators::test_equal PASSED            [  2%]
Dec 11 23:29:33 test/onnx/test_operators.py::TestOperators::test_erf PASSED              [  2%]
Dec 11 23:29:33 test/onnx/test_operators.py::TestOperators::test_exp PASSED              [  2%]
Dec 11 23:29:33 test/onnx/test_operators.py::TestOperators::test_expand PASSED           [  2%]
Dec 11 23:29:33 test/onnx/test_operators.py::TestOperators::test_flatten PASSED          [  2%]
Dec 11 23:29:33 test/onnx/test_operators.py::TestOperators::test_flatten2D PASSED        [  2%]
Dec 11 23:29:33 test/onnx/test_operators.py::TestOperators::test_fmod PASSED             [  2%]
Dec 11 23:29:33 test/onnx/test_operators.py::TestOperators::test_frobenius_norm PASSED   [  2%]
Dec 11 23:29:34 test/onnx/test_operators.py::TestOperators::test_full /var/lib/jenkins/workspace/scripts/onnx/test.sh: line 55: 17688 Segmentation fault      (core dumped) pytest "${args[@]}" -k 'not (TestOperators and test_full_like) and not (TestOperators and test_zeros_like) and not (TestOperators and test_ones_like) and not (TestModels and test_vgg16) and not (TestModels and test_vgg16_bn) and not (TestModels and test_vgg19) and not (TestModels and test_vgg19_bn)' --ignore "$top_dir/test/onnx/test_pytorch_onnx_onnxruntime.py" --ignore "$top_dir/test/onnx/test_custom_ops.py" --ignore "$top_dir/test/onnx/test_models_onnxruntime.py" "${test_paths[@]}"
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3910605) caffe2_onnx_py3_6_clang7_ubuntu16_04_test (3/7)
**Step:** "Test" ([details](https://dr.pytorch.org/build-details.html?build_id=72619478))
```
Dec 11 23:34:55   File "/usr/local/lib/python3.6/dist-packages/pluggy/hooks.py", line 286 in __call__
Dec 11 23:34:55   File "/usr/local/lib/python3.6/dist-packages/_pytest/main.py", line 237 in _main
Dec 11 23:34:55   File "/usr/local/lib/python3.6/dist-packages/_pytest/main.py", line 193 in wrap_session
Dec 11 23:34:55   File "/usr/local/lib/python3.6/dist-packages/_pytest/main.py", line 230 in pytest_cmdline_main
Dec 11 23:34:55   File "/usr/local/lib/python3.6/dist-packages/pluggy/callers.py", line 187 in _multicall
Dec 11 23:34:55   File "/usr/local/lib/python3.6/dist-packages/pluggy/manager.py", line 86 in <lambda>
Dec 11 23:34:55   File "/usr/local/lib/python3.6/dist-packages/pluggy/manager.py", line 92 in _hookexec
Dec 11 23:34:55   File "/usr/local/lib/python3.6/dist-packages/pluggy/hooks.py", line 286 in __call__
Dec 11 23:34:55   File "/usr/local/lib/python3.6/dist-packages/_pytest/config/__init__.py", line 90 in main
Dec 11 23:34:55   File "/usr/local/bin/pytest", line 11 in <module>
Dec 11 23:34:55 /var/lib/jenkins/workspace/scripts/onnx/test.sh: line 55:   453 Segmentation fault      (core dumped) pytest "${args[@]}" -k 'not (TestOperators and test_full_like) and not (TestOperators and test_zeros_like) and not (TestOperators and test_ones_like) and not (TestModels and test_vgg16) and not (TestModels and test_vgg16_bn) and not (TestModels and test_vgg19) and not (TestModels and test_vgg19_bn)' --ignore "$top_dir/test/onnx/test_pytorch_onnx_onnxruntime.py" --ignore "$top_dir/test/onnx/test_custom_ops.py" --ignore "$top_dir/test/onnx/test_models_onnxruntime.py" "${test_paths[@]}"
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3910869) pytorch_linux_xenial_py2_7_9_test (4/7)
**Step:** "Test" ([details](https://dr.pytorch.org/build-details.html?build_id=72626500))
```
Dec 11 23:43:53 Ran 1466 tests in 421.980s
Dec 11 23:43:53 
Dec 11 23:43:53 FAILED (errors=2, skipped=698, expected failures=3)
Dec 11 23:43:53 
Dec 11 23:43:53 Generating XML reports...
Dec 11 23:43:53 Traceback (most recent call last):
Dec 11 23:43:53   File "test/run_test.py", line 455, in <module>
Dec 11 23:43:53     main()
Dec 11 23:43:53   File "test/run_test.py", line 448, in main
Dec 11 23:43:53     raise RuntimeError(message)
Dec 11 23:43:53 RuntimeError: test_nn failed!
Dec 11 23:43:53 =================== sccache compilation log ===================
Dec 11 23:43:53 + cleanup
Dec 11 23:43:53 + retcode=1
Dec 11 23:43:53 + set +x
Dec 11 23:43:53 =========== If your build fails, please take a look at the log above for possible reasons ===========
Dec 11 23:43:53 Compile requests                 10
Dec 11 23:43:53 Compile requests executed         8
Dec 11 23:43:53 Cache hits                        2
Dec 11 23:43:53 Cache misses                      6
Dec 11 23:43:53 Cache timeouts                    0
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3911377) pytorch_macos_10_13_py3_test (5/7)
**Step:** "Test" ([details](https://dr.pytorch.org/build-details.html?build_id=72660112))
```
Dec 11 16:15:22   File "test_fake_quant.py", line 7, in <module>
Dec 11 16:15:22     import hypothesis_utils as hu
Dec 11 16:15:22   File "/Users/distiller/project/test/hypothesis_utils.py", line 9, in <module>
Dec 11 16:15:22     from hypothesis.searchstrategy import SearchStrategy
Dec 11 16:15:22 ModuleNotFoundError: No module named 'hypothesis.searchstrategy'
Dec 11 16:15:22 Traceback (most recent call last):
Dec 11 16:15:22   File "test/run_test.py", line 455, in <module>
Dec 11 16:15:22     main()
Dec 11 16:15:22   File "test/run_test.py", line 448, in main
Dec 11 16:15:22     raise RuntimeError(message)
Dec 11 16:15:22 RuntimeError: test_fake_quant failed!
Dec 11 16:15:22 + cleanup
Dec 11 16:15:22 + retcode=1
Dec 11 16:15:22 + set +x
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3911484) pytorch_linux_xenial_py3_clang5_asan_test (6/7)
**Step:** "Test" ([details](https://dr.pytorch.org/build-details.html?build_id=72660152))
```
Dec 12 00:15:26   File "test_nn.py", line 56, in <module>
Dec 12 00:15:26     import hypothesis_utils as hu
Dec 12 00:15:26   File "/var/lib/jenkins/workspace/test/hypothesis_utils.py", line 9, in <module>
Dec 12 00:15:26     from hypothesis.searchstrategy import SearchStrategy
Dec 12 00:15:26 ModuleNotFoundError: No module named 'hypothesis.searchstrategy'
Dec 12 00:15:26 Traceback (most recent call last):
Dec 12 00:15:26   File "test/run_test.py", line 455, in <module>
Dec 12 00:15:26     main()
Dec 12 00:15:26   File "test/run_test.py", line 448, in main
Dec 12 00:15:26     raise RuntimeError(message)
Dec 12 00:15:26 RuntimeError: test_nn failed!
Dec 12 00:15:26 + cleanup
Dec 12 00:15:26 + retcode=1
Dec 12 00:15:26 + set +x
Dec 12 00:15:26 =================== sccache compilation log ===================
Dec 12 00:15:26 =========== If your build fails, please take a look at the log above for possible reasons ===========
Dec 12 00:15:26 Compile requests                  7
Dec 12 00:15:26 Compile requests executed         6
Dec 12 00:15:26 Cache hits                        0
Dec 12 00:15:26 Cache misses                      6
Dec 12 00:15:26 Cache timeouts                    0
```
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3912487) pytorch_linux_xenial_cuda9_cudnn7_py3_multigpu_test (7/7)
**Step:** "Test" ([details](https://dr.pytorch.org/build-details.html?build_id=72685355))
```
Dec 12 00:54:24 [       OK ] ParallelTest.DataParallelUsesAllAvailableCUDADevices_CUDA (1 ms)
Dec 12 00:54:24 [ RUN      ] ParallelTest.DataParallelNumericalEquivalence_MultiCUDA
Dec 12 00:54:28 [       OK ] ParallelTest.DataParallelNumericalEquivalence_MultiCUDA (3106 ms)
Dec 12 00:54:28 [----------] 9 tests from ParallelTest (7128 ms total)
Dec 12 00:54:28 
Dec 12 00:54:28 [----------] Global test environment tear-down
Dec 12 00:54:28 [==========] 786 tests from 32 test cases ran. (75719 ms total)
Dec 12 00:54:28 [  PASSED  ] 784 tests.
Dec 12 00:54:28 [  FAILED  ] 2 tests, listed below:
Dec 12 00:54:28 [  FAILED  ] IntegrationTest.MNIST_CUDA
Dec 12 00:54:28 [  FAILED  ] IntegrationTest.MNISTBatchNorm_CUDA
Dec 12 00:54:28 
Dec 12 00:54:28  2 FAILED TESTS
Dec 12 00:54:29 + cleanup
Dec 12 00:54:29 + retcode=1
Dec 12 00:54:29 + set +x
Dec 12 00:54:29 =================== sccache compilation log ===================
Dec 12 00:54:29 =========== If your build fails, please take a look at the log above for possible reasons ===========
Dec 12 00:54:30 Compile requests                 0
Dec 12 00:54:30 Compile requests executed        0
Dec 12 00:54:30 Cache hits                       0
```
### 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably caused by upstream breakages:
* [pytorch_xla_linux_xenial_py3_6_clang7_build](https://dr.pytorch.org/build-details.html?build_id=72592543)



---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

VitalyFedyunin committed:
rebase to master on "[WIP] Fix cudnn channels_last descriptio problem"




[ghstack-poisoned]

VitalyFedyunin committed:
Fix flake8 on "[WIP] Fix cudnn channels_last descriptio problem"


Differential Revision: [D18959229](https://our.internmc.facebook.com/intern/diff/D18959229)

[ghstack-poisoned]

VitalyFedyunin committed:
Fix flake8 on "[WIP] Fix cudnn channels_last descriptio problem"


Differential Revision: [D18959229](https://our.internmc.facebook.com/intern/diff/D18959229)

[ghstack-poisoned]

VitalyFedyunin committed:
Add tests on "[WIP] Fix cudnn channels_last descriptio problem"


Differential Revision: [D18959229](https://our.internmc.facebook.com/intern/diff/D18959229)

[ghstack-poisoned]

VitalyFedyunin committed:
experiemnts on "[WIP] Fix cudnn channels_last descriptio problem"


Differential Revision: [D18959229](https://our.internmc.facebook.com/intern/diff/D18959229)

[ghstack-poisoned]
